src/main/java/com/klinik/service/RecordPatientService.java

```diff
@@ -0,0 +1,39 @@
+package com.klinik.service;
+
+import com.klinik.entity.CardPatient;
+import com.klinik.entity.Doctor;
+import com.klinik.entity.RecordPatient;
+import com.klinik.excep.MyException;
+import com.klinik.repositories.CardPatientRepository;
+import com.klinik.repositories.DoctorRerository;
+import com.klinik.repositories.RecordPatientRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class RecordPatientService {
+
+    private final RecordPatientRepository recordPatientRepository;
+    private final DoctorRerository        doctorRerository;
+    private final CardPatientRepository   cardPatientRepository;
+    public List<RecordPatient> allListRecordPatient() {
+        return recordPatientRepository.findAll();
+    }
+    public RecordPatient saveRecordPatient( RecordPatient record_patient, Long doctor_id, Long card_patient_id ) throws Exception{
+        Optional<Doctor> doctor = doctorRerository.findById( doctor_id );
+        if ( doctor.isEmpty() == true ) throw new MyException( 400, "Нет доктора с таким идентификатором");
```

Comment:
можно переписатьиспользуя методы Optional
.ifPresentOrElseThrow() и выкинуть там твой exception.

Reply:
![image](https://github.com/Kirillkuss/rewTwo/assets/99822514/6d0adf7d-080e-4d94-bd8f-bfef942f744a)

Reply:
```suggestion
        if ( doctor.isEmpty() == true ) throw new MyException( 400, "Нет доктора с таким идентификатором");
```
    public RecordPatient saveRecordPatient( RecordPatient recordPatient, Long idDoctor, Long idCardPatient ) throws Exception{
        doctorRerository.findById( idDoctor )
                        .ifPresentOrElse( doc -> { recordPatient.setDoctor( doc );},() ->{new MyException(400, "Нет доктора с таким идентификатором");} );
        Optional<CardPatient> cardPatient = cardPatientRepository.findById( idCardPatient );
        if ( cardPatient.isEmpty() == true ) throw new MyException( 400, "Нет карты пациента с таким идентификатором");
        if ( recordPatientRepository.findById( recordPatient.getIdRecord()).isPresent() == true) throw new MyException( 409, "Запись к врачу с таким ИД уже существует, установите другой ИД записи к врачу");
        recordPatient.setCardPatientId( cardPatient.get().getIdCardPatient() );
        return recordPatientRepository.save( recordPatient );
    }
```

Reply:
`
    public RecordPatient saveRecordPatient( RecordPatient recordPatient, Long idDoctor, Long idCardPatient ) throws Exception{
        doctorRerository.findById( idDoctor )
                        .ifPresentOrElse( doc -> { recordPatient.setDoctor( doc );},() ->{new MyException(400, "Нет доктора с таким идентификатором");} );
        Optional<CardPatient> cardPatient = cardPatientRepository.findById( idCardPatient );
        if ( cardPatient.isEmpty() == true ) throw new MyException( 400, "Нет карты пациента с таким идентификатором");
        if ( recordPatientRepository.findById( recordPatient.getIdRecord()).isPresent() == true) throw new MyException( 409, "Запись к врачу с таким ИД уже существует, установите другой ИД записи к врачу");
        recordPatient.setCardPatientId( cardPatient.get().getIdCardPatient() );
        return recordPatientRepository.save( recordPatient );
    }`

Reply:
.ifPresentOrElseThrow() - разве такой метод есть? не могу найти